src/main/java/edu/uade/apd/tpo/model/Posicion.java

```diff
@@ -93,4 +96,24 @@ public int getNumero() {
     public void setNumero(int numero) {
         this.numero = numero;
     }
+    
+    public void guardar() {
+        PosicionDao.getInstance().save(this);
+    }
+    
+    public void liberar(int cantidad) {
+    	if(cantidad == this.CAPACIDAD) {
+    		this.setEstado(EstadoPosicion.DISPONIBLE);
+    	}
+    	this.setCantidad(this.cantidad - cantidad); 
+    	this.guardar();
+    }
+    
+    public void almacenar(Lote lote, int cantidad, String codigoUbicacion) {
+    	this.setEstado(EstadoPosicion.OCUPADO);
+    	this.setLote(lote);
+    	this.setCodigoUbicacion(codigoUbicacion);
+    	this.setCantidad(cantidad);
+    	this.guardar();
```

Comment:
Eliminar el this.guardar(). Llamar a guardar() desde afuera, en el SistemaDeposito.